docs: improve language of SNP attestation (#1355)
edgelesssys/contrast: docs/docs/architecture/snp.md

# AMD SEV-SNP attestation

This page explains details of the AMD SEV-SNP attestation that are especially relevant for the implementation in Contrast.
You should be familiar with [attestation basics](attestation.md).
For more details on SEV-SNP, see the [whitepapers and specifications from AMD](https://www.amd.com/en/developer/sev.html).

The key component for attesting AMD SEV-SNP is the Secure Processor (SP).
It measures the confidential VM (CVM) and metadata and returns an attestation report reflecting those
measurements.

## Startup

The SP extends the launch digest each time the hypervisor donates a page to the CVM during startup via `SNP_LAUNCH_UPDATE`. On an abstract level, the launch digest is extended as follows:

```
LD := Hash(LD || Page || Page Metadata)
```

When the Hypervisor calls `SNP_LAUNCH_FINISH`, it provides the SP with the `HOST_DATA`,
the `ID_BLOCK`, and `ID_AUTH` block.

The `HOST_DATA` is opaque to the SP. Kata writes the hash of the kata policy in
this field to bind the policy to the CVM. `HOST_DATA` is later reflected in the
attestation report.

### ID block structure

> The ID block defines the identity of the CVM.

Reviewer: I think it makes sense to move the paragraph below the ID block table to here so that the reader knows _how_ the ID block defines the identity (because it's enforced by the SP) and what the "identity" entails (at least the launch digest and the policy). Maybe we can spell that out that this is what we mean by identity. Though, this would be our own concept. Another option is to spell out that the ID block defines the expected contents (via LD) and setting (via policy).

Author: I actually just wanted to give a minimal description of the ID block because the section shouldn't start with "The complete structure ...".
I could replace this sentence with "The ID block contains fields that identify the VM." (derived from a similar sentence in the AMD whitepaper)

We can also go with yours, but I'm not sure if I understood everything. If you think we should go with that, can you phrase the full "ID block structure" section?

Reviewer: > The ID block contains fields that identify the VM.

I like that

Author: ```suggestion
The ID block contains fields that identify the VM.
```
